src/java.base/share/classes/sun/nio/ch/ChannelInputStream.java

```diff
@@ -265,11 +265,43 @@ public long transferTo(OutputStream out) throws IOException {
             return transfer(rbc, fc);
         }
 
+        // ReableByteChannel -> WritableByteChannel
+        if (out instanceof ChannelOutputStream cos) {
```

Comment:
I think it might be better to change the code at L254 to:

```
        if (out instanceof ChannelOutputStream cos) {
            ReadableByteChannel rbc = ch;
            WritableByteChannel wbc = cos.channel();

            // FileChannel -> WritableByteChannel

            // ...
        }
```

That puts the special casing for an output stream backed by a ChannelOutputStream in one place.

Reply:
Agreed and done in 7039304d0ab75fdf7d400a3e8d0f60cd9d535ad6. Funny anecdote: Originally I had it this way but then modified it to simplify the review.